### ISSUE TITLE
Hreflang fix (only self-canonicalized pages must have hreflang)

### ISSUE THREAD

**Issue by rraue** [mozilla/addons-frontend#6842]
### Describe the problem and steps to reproduce it:

Our new hreflang setup has one flaw. Its producing a lot of "no return tag" errors which can confuse the whole setup.
<img width="1779" alt="screen shot 2018-11-06 at 14 32 11" src="https://user-images.githubusercontent.com/36700618/48067541-d521a800-e1d0-11e8-90c5-f764353b7cfd.png">

They appear on pages with a parameter because these pages are not self-canonicalized and tehrefor they get no returning hreflang from the linked pages.

Example: 

/af/firefox/addon/image-search-pro/?src=hotness
canonical: /af/firefox/addon/image-search-pro/
hreflang to: /en-US/firefox/addon/image-search-pro/

/en-US/firefox/addon/image-search-pro/ returns of course only to /af/firefox/addon/image-search-pro/ and therefor the setup is broken.

### What did you expect to happen?

Pages not self canonicalized must not have any hreflang in the head-section. And they do not need it, as a crawler will follow the canonical link in the head and then process the hreflang on the page pointed on by the canonical. Thats why the rule is as well, that no 301 or 404 must be included in any hreflang: it cant return the hreflang link and only a complete redundant hreflang setup garantees to be interpreted correctly.

**Comment by ioanarusiczki** [comment]
@willdurand I verified this issue on AMO dev with FF63 (win10)
The links with `hreflang` are not displayed when the URL contain query parameters. 
I checked with multiple scenarios. Navigating from the detail page to another extension/theme from the More extensions section or while making a search for an extension/theme. However, if the extra query parameters are deleted and page refreshed the `hreflang` links are displayed in the HTML  - head section.

![with and without parameter1](https://user-images.githubusercontent.com/33448286/48143343-3cb22300-e2b7-11e8-8ab0-1055e905b61a.gif)
